### ISSUE TITLE
[feature]: automatic processing of recurring tasks

### ISSUE THREAD

**Issue by imeed166** [therden/packrat#1]
Hi, thank you for the plugin, I've been wanting to do this since first using tasks plugin.
However, I still have to execute the command `Packrat: Process completed recurring Tasks within the active note` manually.

Could you add a way to **periodically** scan notes even if they are **not in the active pane** ? that way, everything would be done automatically. (The interval could be specified in the plugin's settings)

**Comment by therden** [comment]
Hello @imeed166.  Thank you for the suggestion, which I've tagged as a possible enhancement.

I'm not a developer, and had to work hard to figure out enough about the Obsidian API, plugins, and Typescript/javascript generally to get this plugin to the point it's at now.  At present, it is pretty daunting to think about incorporating an always-running loop to periodically fire off a scan of all the markdown files within a vault (which would probably necessitate allowing the user to specify folders to exclude from the scan.)

Ultimately, I hope to convince @schemar to include to add this functionality within the **Tasks**' plugin itself, so that  instances of recurring tasks are archived, moved to the bottom of the note they're in, or deleted _at the time they're completed_ based on the presence of user-definable triggers in the task's description.

**Comment by imeed166** [comment]
I understand, I hope he adds this functionality to the main plugin too.

**Comment by kmaustral** [comment]
I agree, it would be ideal to have this option in the Tasks dialogue itself. I love the idea of Packrat, but the extra command each time adds up. I get virtually the same effect with Task Collector and it removes non-recurring done tasks as well. Keep pushing!

**Comment by therden** [comment]
Hi @kmaustral --Thank you for the encouragement.  **Task Collector** didn't scratch my itch, but I'm glad that it works for you.

I'm curious what about **Packrat** it is that you feel takes too much time.  Is it adding action triggers to the current next/incomplete instances of your recurring tasks (which only needs to be done once per recurring task), or executing **Packrat**'s command (which I generally do only once every few days)?

**Comment by kmaustral** [comment]
The problem is that I have recurring tasks spread over the vault. There is usually one recurring task for each project that schedules time to spend on it.  I have around 30 projects. It would be easier if there was a command to delete all past tasks across the vault.

**Comment by therden** [comment]
Got it; thanks @kmaustral.

It might take a couple of weeks before I have time to work on it, but I _think_ that I'd be able to add a second command to scan all note files in the vault and process any completed recurring tasks according to the Packrat triggers they contain.  

I've created [a new Issue](https://github.com/therden/packrat/issues/1) with your use case, tagged as an Enhancement.

**Comment by blackpeasy** [comment]
Seems like several types of automatic processes are discussed here.

For my own case, the best implementation would be an automatic process that runs -when I mark the task as done-.
I would really love to see the task disappear when I mark it as done, and reappear in an archive note.

**Comment by therden** [comment]
Closing this; various requests addressed in responses to other Issues.
